processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingMethodOptions.java

```diff
@@ -355,6 +355,27 @@ public void applyIgnoreAll(SourceMethod method, TypeFactory typeFactory,
         }
     }
 
+    public void applyIgnoreTargets(SourceMethod method, FormattingMessager messager,
+                                    List<String> ignoreTargets ) {
+        for ( MappingOptions mapping : mappings ) {
+            if ( !mapping.isIgnored() && ignoreTargets.contains( mapping.getTargetName() ) ) {
```

Comment:
I think that we should always report an error if something is defined twice. And the error should be similar to what we have for `PROPERTYMAPPING_DUPLICATE_TARGETS`